### PR TITLE
API Refactor: refactor 'spire' upstream authority to use new API

### PR DESCRIPTION
**Pull request by MarcosDY**
Refactor spire Upstream authority plugin to use new API

### CODE REVIEW

File: pkg/server/plugin/upstreamauthority/spire/spire_server_client.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+		c.conn.Close()
+	}
+	if c.source != nil {
+		c.source.Close()
+	}
+	c.bundleClient = nil
+	c.svidClient = nil
+}
+
+// newDownstreamX509CA request new downstream CAs to server
```
Comment (amartinezfayo) [inline pkg/server/plugin/upstreamauthority/spire/spire_server_client.go:84]:
```suggestion
// newDownstreamX509CA requests new downstream CAs to server
```

File: pkg/server/plugin/upstreamauthority/spire/spire_server_client.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+// newServerClient creates a new spire-sever client
+func newServerClient(serverAddr string, workloadapiSocket string) *serverClient {
+	return &serverClient{
+		serverAddr:        serverAddr,
+		workloadAPISocket: workloadapiSocket,
+	}
+}
+
+type serverClient struct {
```
Comment (amartinezfayo) [inline pkg/server/plugin/upstreamauthority/spire/spire_server_client.go:28]: I'm not sure if serverClient is a good name. I'm wondering if there is a name that better describes the functionality that this struct provides.

Comment (MarcosDY) [reply pkg/server/plugin/upstreamauthority/spire/spire_server_client.go:28]: open to any suggestion

File: pkg/server/plugin/upstreamauthority/spire/spire_server_client.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+// start initializes spire-server endpoints client, it uses X509 source to keep an active connection
+func (c *serverClient) start(ctx context.Context) error {
+	source, err := workloadapi.NewX509Source(ctx, workloadapi.WithClientOptions(workloadapi.WithAddr(c.workloadAPISocket)))
+	if err != nil {
+		return fmt.Errorf("unable to create X509Source: %v", err)
+	}
+
+	tlsConfig := tlsconfig.MTLSClientConfig(source, source, tlsconfig.AuthorizeAny())
```
Comment (azdagron) [inline pkg/server/plugin/upstreamauthority/spire/spire_server_client.go:47]: The peer needs to have the SPIRE server SPIFFE ID (i.e. `tlsconfig.AuthorizeID(serverID)`)

File: pkg/server/plugin/upstreamauthority/spire/spire_server_client.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+// release release connection and clean clients
+func (c *serverClient) release() {
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+
+	if c.conn != nil {
+		c.conn.Close()
+	}
+	if c.source != nil {
+		c.source.Close()
```
Comment (azdagron) [inline pkg/server/plugin/upstreamauthority/spire/spire_server_client.go:78]: should we zero out `c.source` and `c.conn` after closing?

**Pull request merged**